### ISSUE TITLE
PSL coupled_systems backend error

### ISSUE THREAD

**Issue by Himscipy** [pnnl/neuromancer#48]
Hi,
The PSL `coupled_systems.py` is not intergrated with the neuromancer.psl . On an attempts on fixing the issue by modifying the psl `__init__.py` and adding cast_backend decorators to the methods of the coupled system following error is observed.

```
Traceback (most recent call last):
  File "/neuromancer/src/neuromancer/psl/coupled_systems.py", line 449, in <module>
    network = RC_Network.make_5_room(nsim=100)
  File "/neuromancer/src/neuromancer/psl/coupled_systems.py", line 275, in make_5_room
    return RC_Network(nsim=nsim, nx=5, adj=adj)
  File "/neuromancer/src/neuromancer/psl/coupled_systems.py", line 177, in __init__
    super().__init__(nsim, ninit, ts, adj, nx, seed)
  File "/neuromancer/src/neuromancer/psl/coupled_systems.py", line 128, in __init__
    super().__init__(nsim, ninit, ts, seed)
  File "/neuromancer/src/neuromancer/psl/base.py", line 118, in __init__
    self.B = Backend(backend)
  File "/neuromancer/src/neuromancer/psl/base.py", line 63, in __init__
    for k, v in Backend.backends[backend].items():
KeyError: 0
```

**Comment by madelynshapiro** [comment]
Hi @Himscipy, I will look into this. Any example scripts you can share to help reproduce the error would be much appreciated. Thanks for raising this issue!

**Comment by madelynshapiro** [comment]
@Himscipy I believe the source of the `KeyError` can be traced back to the `super().__init__(...)` call in the constructor for `Coupled_NonAutonomous` (which extends `psl.nonautonomous.ODE_NonAutonomous`) 

`coupled_systems.py` lines 124-126:
```python
class Coupled_NonAutonomous(ODE_NonAutonomous):
    def __init__(self, nsim=1001, ninit=0, ts=0.1, adj=None, nx=1, seed=59):
        super().__init__(nsim, ninit, ts, seed)
``` 

This passes `nsim`, `ninit`, `ts`, `seed` as positional arguments to the `ODE_NonAutonomous` constructor, the signature of which was changed in the recent refactor of PSL. `ODE_NonAutonomous` currently uses the default constructor signature defined by `EmulatorBase`.

`base.py` lines 110 - 120:
```python
class EmulatorBase(ABC, torch.nn.Module):
    def __init__(self, exclude_norms=['Time'], backend='numpy', requires_grad=False,
                 seed: Union[int,np.random._generator.Generator]=59, set_stats=True):
        """

        :param ts: (float) Time step for numerical integration
        :param seed: (int) Fixes random seed
        :param exclude_norms: (List of str) Keys to any data that should not be normalized such as 'Time'
        :param backend: (str) Can be 'torch' or 'numpy'
        :param requires_grad: (bool) Set this to true for parameter tuning with a pytorch backend
        """
``` 

Try replacing line 126 of `coupled_systems.py` with the following:

```diff
-        super().__init__(nsim, ninit, ts, seed)
+        self.nsim = nsim
+        self.ninit = ninit
+        self.ts = ts
+        super().__init__(seed=seed)
```

If this resolves your issue, please submit a PR including the bugfix, modifications to `__init__.py`, and additional `@cast_backend` decorators. If not, please follow up with any new details. Thank you! 😃

**Comment by madelynshapiro** [comment]
@Himscipy Closing this issue for now, we believe this issue has been fixed by #64 . If the error persists on your end, please feel free to reopen the issue with some example code that can be used to reproduce the error. Thank you!
